coordinator/src/main/java/org/apache/uniffle/coordinator/ApplicationManager.java

```diff
@@ -152,6 +154,50 @@ public void refreshAppId(String appId) {
     }
   }
 
+  public void refreshRemoteStorages(ClientConf dynamicClientConf) {
+    if (dynamicClientConf == null || MapUtils.isEmpty(dynamicClientConf.getRemoteStorageInfos())) {
+      LOG.info("Refresh remote storage with empty config");
+      for (String path : availableRemoteStorageInfo.keySet()) {
+        removePathFromCounter(path);
+      }
+      availableRemoteStorageInfo.clear();
+      return;
+    }
+
+    Map<String, RemoteStorageInfo> remoteStorageInfoMap = dynamicClientConf.getRemoteStorageInfos();
+    LOG.info("Refresh remote storage with {}", remoteStorageInfoMap);
+
+    for (Map.Entry<String, RemoteStorageInfo> entry : remoteStorageInfoMap.entrySet()) {
+      String path = entry.getKey();
+      RemoteStorageInfo rsInfo = entry.getValue();
+
+      if (!availableRemoteStorageInfo.containsKey(path)) {
+        remoteStoragePathRankValue.computeIfAbsent(
+            path,
+            key -> {
+              // refreshRemoteStorage is designed without multiple thread problem
+              // metrics shouldn't be added duplicated
+              addRemoteStorageMetrics(path);
+              return new RankValue(0);
+            });
+      }
+      availableRemoteStorageInfo.put(path, rsInfo);
+    }
+    // remove unused remote path if exist
+    List<String> unusedPath = Lists.newArrayList();
+    for (String existPath : availableRemoteStorageInfo.keySet()) {
+      if (!remoteStorageInfoMap.keySet().contains(existPath)) {
+        unusedPath.add(existPath);
+      }
+    }
+    // remote unused path
+    for (String path : unusedPath) {
+      availableRemoteStorageInfo.remove(path);
+      // try to remove if counter = 0, or it will be removed in decRemoteStorageCounter() later
+      removePathFromCounter(path);
+    }
+  }
```

Comment:
```java
    // remove unused remote path if exist
    List<String> unusedPath = Lists.newArrayList();
    for (String existPath : availableRemoteStorageInfo.keySet()) {
      if (!remoteStorageInfoMap.keySet().contains(existPath)) {
        unusedPath.add(existPath);
      }
    }
    // remote unused path
    for (String path : unusedPath) {
      availableRemoteStorageInfo.remove(path);
      // try to remove if counter = 0, or it will be removed in decRemoteStorageCounter() later
      removePathFromCounter(path);
    }
```
Perhaps these two loops can be merged into one without the need for an intermediate structure unusedPath like

```java
Iterator<String> iterator = availableRemoteStorageInfo.keySet().iterator();
while (iterator.hasNext()) {
    String path = iterator.next();
    if (!remoteStorageInfoMap.containsKey(path)) {
        iterator.remove();
        removePathFromCounter(path);
    }
}
```